modules/shared/src/main/scala/org/tessellation/schema/snapshot.scala

```diff
@@ -14,8 +14,30 @@ import org.tessellation.schema.transaction.{Transaction, TransactionReference}
 import org.tessellation.security.hash.Hash
 import org.tessellation.syntax.sortedCollection._
 
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.string.MatchesRegex
+import io.circe.refined._
+import io.estatico.newtype.macros.newtype
+
 object snapshot {
 
+  @derive(encoder, decoder, eqv, show)
+  @newtype
+  case class SemVer(
```

Comment:
Maybe it would be a good idea to make the `SemVer` an ADT? Every new version that's gonna be released will be a new case object added to the family? 🤔

Reply:
something like:
```
  @derive(encoder, decoder, eqv, show)
  sealed trait SemVer {
    val version: String Refined MatchesRegex[...]
  }

  object SemVer {
    case object `0.0.1` extends SemVer { val version = "0.0.1" }
  }
```

Reply:
In case we won't do anything in our codebase based on these versions we won't need to keep the history of them as ADT as well. Just a hardcoded version field for a snapshot should be enough. We need to think about the exact use case for this to really decide.